scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/Problem.java

```diff
@@ -1,48 +1,99 @@
 package gov.nasa.jpl.aerie.scheduler;
 
+import gov.nasa.jpl.aerie.merlin.driver.MissionModel;
+
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 /**
  * description of a planning problem to be solved
  */
 public class Problem {
 
   /**
-   * creates a new empty problem based in the given mission model
-   *
-   * @param mission IN the mission model that this problem is based on
+   * the mission model that this problem is based on
    */
-  public Problem(MissionModelWrapper mission) {
-    if (mission == null) {
-      throw new IllegalArgumentException(
-          "creating problem descriptor with null mission model");
-    }
-    this.mission = mission;
-    this.initialPlan = new PlanInMemory(mission);
-  }
+  private final MissionModel<?> missionModel;
+
+  private final SimulationFacade simulationFacade;
+
+  //should be accessible inside problem definition
+  protected final PlanningHorizon planningHorizon;
 
   /**
-   * the mission model that this problem is based on
+   * global constraints in the mission model, indexed by name
    */
-  protected MissionModelWrapper mission;
-
+  private final List<GlobalConstraint> globalConstraints
+      = new java.util.LinkedList<>();
   /**
    * the initial seed plan to start scheduling from
    */
-  protected Plan initialPlan;
+  private Plan initialPlan;
 
   /**
    * container of all goals in the problem, indexed by name
    */
-  protected final java.util.HashMap<String, Goal> goalsByName = new java.util.HashMap<>();
+  private final java.util.HashMap<String, Goal> goalsByName = new java.util.HashMap<>();
+
+  /**
+   * activity type definitions in the mission model, indexed by name
+   */
+  private final java.util.Map<String, ActivityType> actTypeByName
+      = new java.util.HashMap<>();
+
+  /**
+   * creates a new empty problem based in the given mission model
+   *
+   * @param mission IN the mission model that this problem is based on
+   */
+  public Problem(MissionModel<?> mission, PlanningHorizon planningHorizon) {
+    this.missionModel = mission;
+    this.initialPlan = new PlanInMemory();
+    this.planningHorizon = planningHorizon;
+    //TODO: find cleaner way to handle built-in act types
```

Comment:
:large_orange_diamond: Can you fill me in about what function is performed by these built-in activity types?